python/ray/dag/compiled_dag_node.py

```diff
@@ -709,8 +709,17 @@ def _get_or_compile(
                 # in the DAG.
                 readers = [self.idx_to_task[idx] for idx in task.downstream_node_idxs]
 
-                if isinstance(readers[0].dag_node, MultiOutputNode):
-                    assert len(readers) == 1
+                dag_nodes = [reader.dag_node for reader in readers]
+                has_multi_output_node = False
+                for dag_node in dag_nodes:
+                    if isinstance(dag_node, MultiOutputNode):
+                        has_multi_output_node = True
+                        break
+                if has_multi_output_node:
+                    if len(readers) != 1:
+                        raise ValueError(
+                            "Only the driver may read from a MultiOutputNode."
```

Comment:
```suggestion
                            "DAG outputs currently cannot be read by other actor tasks."
```

Reply:
(technically the other actors are not reading the MultiOutputNode - they are reading from *one* of the nodes marked as an output)

Reply:
Nice catch! Done